scrapy/downloadermiddlewares/httpproxy.py

```diff
@@ -9,11 +9,13 @@
 
 from scrapy.utils.httpobj import urlparse_cached
 from scrapy.exceptions import NotConfigured
+from scrapy.utils.python import to_bytes
 
 
 class HttpProxyMiddleware(object):
 
-    def __init__(self):
+    def __init__(self, auth_encoding=None):
```

Comment:
Yeah I think it should have 'latin-1' as it did before. I just added a default for backwards compatibility as this middleware could be being used without arguments. (like in the tests https://github.com/scrapy/scrapy/blob/master/tests/test_downloadermiddleware_httpproxy.py#L28)